### PR TITLE
[router] Cleanup per storage node router quota code

### PR DESCRIPTION
**Pull request by majisourav99**
## [router] Cleanup per storage node router quota code

Since per store router quota is deployed in all fabrics, this PR cleans up per storage node quota code and associated tests.
## How was this PR tested?

GHCI
## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

**Review by gaojieliu — CHANGES_REQUESTED** [review]

File: services/venice-router/src/main/java/com/linkedin/venice/router/throttle/ReadRequestThrottler.java [hunk 1]
```diff
@@ -102,8 +98,7 @@ public ReadRequestThrottler(
       double perStorageNodeReadQuotaBuffer,
       double perStoreRouterQuotaBuffer,
       long storeQuotaCheckTimeWindow,
-      long storageNodeQuotaCheckTimeWindow,
-      boolean perStorageNodeThrottlerEnabled) {
+      long storageNodeQuotaCheckTimeWindow) {
```
[the remark this answers is not in the document]
Comment (majisourav99) [reply services/venice-router/src/main/java/com/linkedin/venice/router/throttle/ReadRequestThrottler.java:95]: you mean `perStorageNodeReadQuotaBuffer` ? I can remove it

File: services/venice-router/src/main/java/com/linkedin/venice/router/throttle/ReadRequestThrottler.java [hunk 2]
```diff
@@ -264,18 +239,6 @@ public void handleRouterCountChanged(int newRouterCount) {
 
   @Override
   public void onExternalViewChange(PartitionAssignment partitionAssignment) {
```
[the remark this answers is not in the document]
Comment (majisourav99) [reply services/venice-router/src/main/java/com/linkedin/venice/router/throttle/ReadRequestThrottler.java]: Sure

File: services/venice-router/src/main/java/com/linkedin/venice/router/throttle/ReadRequestThrottler.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (gaojieliu) [reply services/venice-router/src/main/java/com/linkedin/venice/router/throttle/ReadRequestThrottler.java:95]: Also `storageNodeQuotaCheckTimeWindow`.

File: services/venice-router/src/main/java/com/linkedin/venice/router/throttle/ReadRequestThrottler.java [hunk 3]
```diff
@@ -86,11 +78,8 @@ public ReadRequestThrottler(
         routingDataRepository,
         routerConfig.getMaxReadCapacityCu(),
         stats,
-        routerConfig.getPerStorageNodeReadQuotaBuffer(),
```
Comment (gaojieliu) [inline services/venice-router/src/main/java/com/linkedin/venice/router/throttle/ReadRequestThrottler.java:89]: Can we get rid of this config: `getPerStorageNodeReadQuotaBuffer` from VeniceRouterConfig as well? Also the corresponding config entries in `ConfigKeys`?

**Review by gaojieliu — APPROVED** [review]

**Pull request merged**